packages/api/src/command/pagination.ts

```diff
@@ -0,0 +1,42 @@
+import { FindOptions, Op, OrderItem } from "sequelize";
+import { UnionToIntersection, XOR } from "ts-essentials";
+
+export type Pagination = XOR<PaginationFromItem, PaginationToItem> & {
+  /** indicates the number of items to be included in the response */
+  count: number;
+};
+
+export type PaginationFromItem = {
+  /** indicates the minimum item to be included in the response, inclusive */
+  fromItem?: string | undefined;
+};
+export type PaginationToItem = {
+  /** indicates the maximum item to be included in the response, inclusive */
+  toItem?: string | undefined;
+};
+export type PaginationItem = UnionToIntersection<PaginationFromItem | PaginationToItem>;
+
+export function getPaginationFilters(pagination: Pagination | undefined) {
+  const { toItem, fromItem } = pagination ?? {};
+  return {
+    ...(toItem ? { id: { [Op.lte]: toItem } } : undefined),
```

Comment:
do we prefer:

`...(toItem && { id: { [Op.lte]: toItem } })` or `...(toItem ? { id: { [Op.lte]: toItem } } : undefined`

they have slightly different effects the input is boolean but we should maybe pick a path in my mind.

Reply:
I don't like this...

```ts
...(toItem && { id: { [Op.lte]: toItem } })
```

...because it results in the value of `toItem`, which can be undefined but can also be `false`.

I try to only have the first one on `.tsx` files, but I'd prefer to be more explicit (second form).